src/components/user-avatar.test.tsx

```diff
@@ -0,0 +1,41 @@
+import { describe, expect, it, vi } from 'vitest'
+import { render, screen, waitFor } from '@testing-library/react'
+import { renderWithProviders } from '@/tests/unit.helpers'
+import { useUser } from '@clerk/nextjs'
+import { UseUserReturn } from '@clerk/types'
+import { UserAvatar } from '@/components/user-avatar'
+
+vi.mock('@mantine/core', () => ({
+    Avatar: ({ src, alt }: { src?: string; alt?: string }) => (
+        <img data-testid="avatar-component" src={src} alt={alt} />
+    ),
+}))
+
+describe('UserAvatar', () => {
+    it('displays the user profile image when logged in', async () => {
+        vi.mocked(useUser).mockReturnValue({
+            user: {
+                profileImageUrl: 'https://example.com/profile.jpg',
+            },
+        } as UseUserReturn)
+
```

Comment:
we have mocks for clerk already setup for all tests in `tests/vitest.setup.ts`, and also have a lot of ways to mock the current user, see `tests/unit.helpers.tsx`, but I do not see where we're returning a profileImageUrl from the `mockClerkSession`.   Adding that seems like a natural addition though, can you add it there and then use it in this test?